test/LOAuth.test.ts

```diff
@@ -95,219 +101,271 @@ for (const param of ctorRequired) {
+    }
+  ];
+  for (const config of startAutomaticTokenRefreshConfigs) {
+    test(config.testTitle, async () => {
+      auth.refreshAccessToken = jest.fn();
+      const options = config.override ? {
+        interval: config.interval,
+        refreshWindow: config.refreshWindow
+      } : {};
+      await auth.startAutomaticTokenRefresh(options);
+      mockToken.expires = new Date(Date.now() + ((config.refreshWindow + 2) * 60 * 1000));
+      auth.token = mockToken;
+
+      expect(auth.refreshAccessToken.mock.calls.length).toBe(1); // Initial retrieval of token from URL
+      expect(globals.window.setInterval.mock.calls.length).toBe(1);
+      expect(globals.window.setInterval.mock.calls[0][1]).toBe(config.interval);
+      // Mock setInterval func invocation
+      globals.window.setInterval.mock.calls[0][0].bind(auth)();
+      expect(auth.refreshAccessToken.mock.calls.length).toBe(1); // Hasn't expired yet
+
+      mockToken.expires = new Date(Date.now() + ((config.refreshWindow - 1) * 60 * 1000));
+      globals.window.setInterval.mock.calls[0][0].bind(auth)();
+      expect(auth.refreshAccessToken.mock.calls.length).toBe(2); // Expired and called
+      expect(auth.refreshAccessToken.mock.calls[1][0]).toEqual({ expiringRefresh: true });
+    });
+  }
 
-test('sign uses client-oauth2 sign if token present', async () => {
-  auth.token = mockToken;
-  const requestOptions = {
-    method: 'GET',
-    url: 'https://integ-test/api'
-  };
-  await auth.sign(requestOptions);
-  expect(mockToken.sign.mock.calls.length).toBe(1);
-  expect(mockToken.sign.mock.calls[0][0]).toEqual(requestOptions);
-});
+  test('stopAutomaticTokenRefresh calls clearInterval if refreshInterval is set', async () => {
+    await auth.stopAutomaticTokenRefresh();
+    expect(globals.window.clearInterval.mock.calls.length).toBe(0);
 
-test('sign throws if not token captured', async () => {
-  auth.token = null;
-  const requestOptions = {
-    method: 'GET',
-    url: 'https://integ-test/api'
-  };
-  expect(auth.sign(requestOptions)).rejects.toThrow();
-});
+    auth.refreshInterval = () => {};
 
-test('logout redirects to the logoutUri', async () => {
-  auth.stopAutomaticTokenRefresh = jest.fn();
-  await auth.logout();
+    await auth.stopAutomaticTokenRefresh();
+    expect(globals.window.clearInterval.mock.calls.length).toBe(1);
+    expect(globals.window.clearInterval.mock.calls[0][0]).toBe(auth.refreshInterval);
+  });
 
-  expect(auth.stopAutomaticTokenRefresh.mock.calls.length).toBe(1);
-  const qs = queryString.stringify({
-    client_id: ctorParams.clientId,
-    logout_uri: ctorParams.logoutRedirectUri
+  test('sign uses client-oauth2 sign if token present', async () => {
+    auth.token = mockToken;
+    const requestOptions = {
+      method: 'GET',
+      url: 'https://integ-test/api'
+    };
+    await auth.sign(requestOptions);
+    expect(mockToken.sign.mock.calls.length).toBe(1);
```

Comment:
no change needed, but jsut a heads up jest has helpers for theses `expect(fn).toBeCalled()` or `expect(fn).toBeCalledTimes(1)` also there is a helper for validating args: 
```
expect(fn).toBeCalledWith(
  requestOptions,
  expect.any(String), // any string
  expect.anything() // anything
);

Reply:
Oo nice- thanks for the note.  I've got a follow up PR coming with several cleanup items; I'll include this.